### ISSUE TITLE
Admin\Contacts - External User can add a contact

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1351]
#### Description:
As an External User, I want the ability to add new contacts

[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?node-id=8953-2004&t=nbQdimhK7lE57KVd-4)

#### Acceptance Criteria:
Given I am an authenticated user with the `industry-user-admin` role (i.e., I have been Approved to have `admin` rights for an Operator)
When I navigate to the "Contacts" table
Then I can add a new contact

Given that I am on the contacts page
When I click the button add a new contact
Then it will open a form where I can fill out the details of the contact
https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=8953-2004&t=rBpJCzhxRzIMrjwm-4

Given that I am on the add contacts page
When I selected Yes to "Is the operation representative a BCIERS user" 
Then their information will be pre-populated instead of manually filling it out again

Given that I'm on the add contacts page
When I click submit
Then I can see a confirmation that this contact has been successfully added attn: @nanyangpro 

#### Development Checklist:

- [ ] 

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies
Blocking #1607

**Comment by patrickisaac** [comment]
Notes:
- Pat to sync with Zoey on refining/separating these into three tickets after #1350 has been revised

**Comment by zoeyli-46** [comment]
@nanyangpro for the user roles, do industry admin and regular industry user both have the permissions to edit add and delete contacts? Or just the admin

**Comment by Sepehr-Sobhani** [comment]
@andrea-williams What will the 'business_role' be for the newly created contact? This is a mandatory field in the Contact table.

**Comment by andrea-williams** [comment]
@Sepehr-Sobhani  hmmm I think we'll want to replace that field in the Contact table to make it nullable. But we do also have "Operation Representative" as an existing option in the `business_role` table, so that can be used.

**Comment by Sepehr-Sobhani** [comment]
@andrea-williams Thanks for the quick response. For now, I'm going to use "Operation Representative" as it will be a safer option.

**Comment by shon-button** [comment]
@patriciarussellCAS 

**Reachable at:**
 https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/contacts/add-contact

**Comment by patriciarussellCAS** [comment]
Link for this URL is not working in Test.

**Comment by patriciarussellCAS** [comment]
I'm getting the following message in Dev @shon-button @Sepehr-Sobhani 

![Image](https://github.com/user-attachments/assets/c4811566-71dc-4747-8377-be0f8b55bdef)

**Comment by Sepehr-Sobhani** [comment]
> I'm getting the following message in Dev @shon-button @Sepehr-Sobhani
> 
> ![Image](https://private-user-images.githubusercontent.com/94197323/365260082-c4811566-71dc-4747-8377-be0f8b55bdef.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NDU0MjEsIm5iZiI6MTcyNTY0NTEyMSwicGF0aCI6Ii85NDE5NzMyMy8zNjUyNjAwODItYzQ4MTE1NjYtNzFkYy00NzQ3LTgzNzctYmUwZjhiNTViZGVmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDE3NTIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2ZThiMDg2YTNkNGYxMDkwZDFkM2UwMGMyYmM5YmI3MWY4NDI0NjQ1ZmFmOWYyNGM4MjdjNGQ1ZDcwYjY3NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7HE29iEQCzQpxgmw_31uh4sRKcKfU7-ThFFUhAw80Xc)

I wasn't able to reproduce this, Can we walk through this feature together?

**Comment by patriciarussellCAS** [comment]
@Sepehr-Sobhani I logged out and logged back in and it seems to work now! ignore my comment above. thanks!

**Comment by patriciarussellCAS** [comment]
Review notes: 
- Could not review this in test, as the "add contact" button was not included. Works fine in Dev. 
- If we are keeping the Y/N "Is this contact a user in BCIERS?" then I noticed the following (please ignore if this is no longer relevant): 
- In Dev, on the Add Contact page, if I select a user from the drop-down (i.e. Yes, they are already a user in BCIERS) and then select a name, the information auto-populates on the page (yay!). But if i then Deselect "is this contact a user in BCIERS?" (i.e select No) - that person I previously selected stayed auto-filled in the fields below.  If this feature is being kept in the system, in future we may want the selection of "no" to clear the fields below.
